client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -195,6 +196,26 @@ export class BlockEditorEditingPresentational extends React.Component<
     }
   };
 
+  private handleSquareBracketExport() {
+    // check to see whether there are a balanced number of square brackets
+    // if there are, everything within the outermost brackets gets exported
+    const didConvertBrackets = slateChangeMutations.scanBlockAndConvertOuterSquareBrackets({
+      change: this.props.value.change(),
+      updateBlock: this.props.updateBlock,
+      exportSelection: this.props.exportSelection,
+      blockId: this.props.block.id,
+    });
+
+    // if something was exported, redo this process
+    if (didConvertBrackets) {
+      setTimeout(() => this.handleSquareBracketExport(), 10);
```

Comment:
I don't believe you need that inner function if ``handleSquareBracketExport`` doesn't take any args. You can just write it as (I think),
```js
 setTimeout(this.handleSquareBracketExport, 10);
```

Reply:
Have you thought much about ways to do this without the recursion here? It seems pretty messy to me, though I'd have to spend some time with it to get a better sense.

Reply:
Yeah I explored a few different approaches. This is definitely the most straightforward and least brittle. The central issue is that, given the way that app is currently structured, exports happen one at a time, and exports are based on a selection. So when you have multiple exports that you want to do, you run into the following difficulty. After the first exports occurs (since they have to happen one at a time), the structure of the document has changed, and all of your previous selection information is obsolete. This makes it really tricky to do anything other than the (kinda awkward) approach taken here: perform one export and then take a look at the new document to get the selection info for the next export.